File: jetpack/lib-arch/src/main/java/com/yyxnb/oh/arch/lifecycle/IFragmentLifecycle.java
package com.yyxnb.oh.arch.lifecycle;

import android.content.Context;
import android.os.Bundle;
import android.view.View;

import com.yyxnb.oh.core.interfaces.ILifecycle;

/**
 * ================================================
 * 作    者：yyx
 * 版    本：1.0
 * 日    期：2020/11/21
 * 历    史：
 * 描    述：IFragmentDelegate
 * ================================================
 */
public interface IFragmentLifecycle extends ILifecycle {

    void onAttached(Context context);

    void onCreated(Bundle savedInstanceState);

    void onViewCreated(View view, Bundle savedInstanceState);

    void onActivityCreated(Bundle savedInstanceState);

    void onStarted();

    void onResumed();

    void onPaused();

    void onStopped();

    void onSaveInstanceState(Bundle outState);

    void onViewDestroyed();

    void onDestroyed();

    void onDetached();

    boolean isAdd();

}

File: lib_arch/src/main/java/com/yyxnb/arch/interfaces/IFragmentLifecycle.kt
package com.yyxnb.arch.interfaces

import android.content.Context
import android.os.Bundle
import android.view.View

/**
 * fragment 代理接口
 */
interface IFragmentLifecycle {

    fun onAttached(context: Context)

    fun onCreated(savedInstanceState: Bundle?)

    fun onViewCreated(v: View, savedInstanceState: Bundle?)

    fun onActivityCreate(savedInstanceState: Bundle?)

    fun onStarted()

    fun onResumed()

    fun onPaused()

    fun onStopped()

    fun onSaveInstanceState(outState: Bundle)

    fun onViewDestroyed()

    fun onDestroyed()

    fun onDetached()

    fun isAdded(): Boolean
}